### PR TITLE
Use sphinx-book-theme

### PR DESCRIPTION
**Pull request by mathbunnyru**
Sorry for creating new PR, I merged docs environment to make this PR easier to follow.

Fix: https://github.com/jupyter/docker-stacks/issues/1597

### CODE REVIEW

**Comment by mathbunnyru** [comment]
Comment by @consideRatio 

@choldgraf i think you overview the options well, do you have suggestions on a theme to use for a jupyter project in the switch from alabaster?

**Comment by mathbunnyru** [comment]
Now, changes to docs are small and easy and don't trigger images rebuild.

**Comment by choldgraf** [comment]
Regarding themes, I think that the two relevant themes are:

- [sphinx-book-theme](https://sphinx-book-theme.readthedocs.io/) - what Jupyter Book uses and several other projects in the Jupyter ecosystem
- [pydata sphinx theme](https://pydata-sphinx-theme.readthedocs.io/) - has a topbar and is probably better-suited to more complex website layouts

I think either of these would be fine and also give some consistency across the Jupyter community. I think the question to answer is whether you think there are multiple user personas for the docs that are different enough that they warrant different "top-level sections". If that's the case, perhaps the pydata theme makes sense since it's got the topbar for navigation. Otherwise, the book theme is a simpler layout

**Comment by mathbunnyru** [comment]
Bad news:
[sphinx-book-theme](https://sphinx-book-theme.readthedocs.io/) will not work for us right now.

Good news:
It should work pretty soon

The thing is we need to use modern docutils `>=0.17` for the parser option https://myst-parser.readthedocs.io/en/latest/sphinx/use.html#include-markdown-files-into-an-rst-file.

The theme doesn't support it yet:
https://github.com/executablebooks/sphinx-book-theme/issues/392
But there is a PR:
https://github.com/executablebooks/sphinx-book-theme/pull/472

So, let's wait the merge of this PR and then we will be able to use sphinx-book-theme.

**Comment by choldgraf** [comment]
I'll get that PR in ASAP :-)

File: .github/workflows/sphinx.yml
Comment (consideRatio) [inline .github/workflows/sphinx.yml:37]:
```suggestion
          python-version: "3.9"
```

When specifyinh version 3.10, not using quotes mean we get version 3.1. lets avoid that ahead of time by providing quotes.

Comment (mathbunnyru) [reply .github/workflows/sphinx.yml]: Didn't know about this, thank you!

**Comment by mathbunnyru** [comment]
@consideRatio @trallard @romainx @parente I think this theme works very well for us and this makes our docs look modern.
Check it out: https://jupyter-docker-stacks--1602.org.readthedocs.build/en/1602/index.html

@choldgraf I would like to thank you for this theme - it's a great work.

The only thing that didn't work for me is fullscreen mode in Safari (desktop of course).
I created an issue for this.
But it's not the release problem, I think it has never worked.
https://github.com/executablebooks/sphinx-book-theme/issues/477

**Comment by mathbunnyru** [comment]
This a draft for now - waiting for a release.

**Comment by trallard** [comment]
I love the new look and it seems more in line with other projects in Jupyter. 

Question: all pages have a warning saying that they were created from a PR and all point to the same PR number. Will this be kept in the future?

**Comment by mathbunnyru** [comment]
Nope:) It happens, because it was generated from this PR. When I merge to master, everything will be nice and clear :)

**Comment by consideRatio** [comment]
Okay we are awaiting a release of sphinx-book-theme more modern than 0.2.0!

See https://github.com/executablebooks/sphinx-book-theme/tags for updates. Thanks @mathbunnyru, @choldgraf, and @trallard!

**Review by consideRatio — APPROVED** [review]
LGTM for merge now or after a release of sphinx-book-theme versioned > 0.2.0 (see https://github.com/executablebooks/sphinx-book-theme/tags to check for new releases).

**Comment by mathbunnyru** [comment]
I decided to merge this with pinned commit (the latest in master at the moment).
If there is something wrong with the docs, we will receive the feedback and will ask the upstream project to fix it.
I spent some time playing with the new docs - found only one issue, everything else is fine.

Also subscribed to releases, so will switch to release when it is available.

**Pull request merged**
